src/path.jl

```diff
@@ -253,7 +252,7 @@ julia> join(p"~/.julia/v0.6", "REQUIRE")
 p"~/.julia/v0.6/REQUIRE"
 ```
 """
-function Base.join(prefix::T, pieces::Union{AbstractPath, AbstractString}...) where T <: AbstractPath
+function join(prefix::T, pieces::Union{AbstractPath, AbstractString}...) where T <: AbstractPath
```

Comment:
Why make this method parametric?
```suggestion
function join(prefix::AbstractPath, pieces::Union{AbstractPath, AbstractString}...)
```

Reply:
Legacy reasons for it were because we used to construct the new path with `T(...)`. That's not needed anymore.